src/Utilities/sharedFunctions.js

```diff
@@ -0,0 +1,19 @@
+export const subtractWeeks = (numOfWeeks, date = new Date()) => {
+    date.setDate(date.getDate() - numOfWeeks * 7);
+
+    return date;
+};
+
+export const verifyDisconnectedSystem = (perReporterStaleness) => {
+    if (perReporterStaleness && !perReporterStaleness.puptoo) {
+        return true;
+    } else if (perReporterStaleness && perReporterStaleness.puptoo) {
+        const stalenessDate = new Date(perReporterStaleness.puptoo?.stale_timestamp);
```

Comment:
Is it necessary to use `?.` here? Just wondering since you confirmed the presence of `perReporterStaleness.puptoo` in the conditional